### PR TITLE
feat(content_service_improvement)

### PR DESCRIPTION
**Pull request by vjaouen**
: improve first content service implementation :

* DeletablePathResolver --> ResourceResolver
* ContentServiceImpl 100% covered by unit tests

**Please check if the PR fulfills these requirements**

- [X] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format) ?
- [X] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [X] The code coverage on new code >75%
- [X] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [X] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Does this PR introduce a breaking change?**
- [ ] Yes
- [X] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: daikon-content-service/content-service-common/src/test/java/org/talend/daikon/content/ContentServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        final String location = "tete";
+        when(resourceResolver.getResource(location)).thenReturn(resource);
+        doThrow(new IOException("error")).when(resource).delete();
+
+        // when
+        service.delete(location);
+    }
+
+    @Test
+    public void shouldClear() throws Exception {
```
Comment (vjaouen) [inline daikon-content-service/content-service-common/src/test/java/org/talend/daikon/content/ContentServiceImplTest.java]: The Mocktio verify method is called. Codacy should be updated with a new rule for mockito based unit tests.

File: daikon-content-service/content-service-common/src/test/java/org/talend/daikon/content/ContentServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        final DeletableResource deletableResource= mock(DeletableResource.class);
+        when(resourceResolver.getResource(location)).thenReturn(deletableResource);
+        when(deletableResource.getInputStream()).thenThrow(new IOException("error"));
+
+        // when
+        service.put(location);
+    }
+
+    @Test
+    public void shouldDeleteResource() throws Exception {
```
Comment (vjaouen) [inline daikon-content-service/content-service-common/src/test/java/org/talend/daikon/content/ContentServiceImplTest.java]: The Mocktio verify method is called. Codacy should be updated with a new rule for mockito based unit tests.

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/177/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/177/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/178/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/178/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/181/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/181/)

**Pull request merged**